### PR TITLE
fix: keycloak operator retry

### PR DESCRIPTION
**Pull request by merll**
This PR changes the error handling in the Keycloak operator with minimal changes to the functionality, for having the operator retry on failures that were sometimes only logged to the console. This includes the following changes:
* Errors are in most cases no longer handled gracefully within a function, but instead thrown into the outer call of `retryOperation`, which reattempts recalling them indefinitely (that part has not been altered).
* `retryOperation` has been modified to re-attempt in a loop instead of calling itself recursively.
* Error handling has been moved to a separate function `extractError` which wraps errors in `WrappedError` to avoid duplicate messages while escalating to `retryOperation`. In addition, it extracts only the relevant information from HttpErrors to avoid leaking excess data (e.g. auth headers) to the console.
* Generally some function calls have been flattened.

### CODE REVIEW

**Review by ferruhcihan — APPROVED** [review]
Tested with a cluster. It works as expected. 👍

File: src/tasks/keycloak/errors.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { HttpError as K8sHttpError } from '@kubernetes/client-node'
+import { HttpError as KeyCloakHttpError } from '@linode/keycloak-client-node'
+
+export class WrappedError extends Error {}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function extractError(operationName: string, error: any): any {
```
Comment (CasLubbers) [inline src/tasks/keycloak/errors.ts:7]: Did using the Error object like this not work? Then we could also remove the eslint line above. And the less `any` the better :) 
```suggestion
export function extractError(operationName: string, error: Error): Error {
```

**Review by CasLubbers — APPROVED** [review]
One small remark

**Review by ElderMatt — APPROVED** [review]

File: src/tasks/keycloak/errors.ts (hunk 1 above)
Comment (merll) [reply src/tasks/keycloak/errors.ts]: Good suggestion, thanks!

**Pull request merged**